python/sglang/srt/models/deepseek_v2.py

```diff
@@ -209,13 +210,21 @@ def __init__(
             )
         self.act_fn = SiluAndMul()
 
-    def forward(self, x, forward_batch=None, can_fuse_mlp_allreduce=False):
+    def forward(
+        self,
+        x,
+        forward_batch=None,
+        can_fuse_mlp_allreduce: bool = False,
+        use_reduce_scatter: bool = False,
+    ):
         if (self.tp_size == 1) and x.shape[0] == 0:
             return x
 
         gate_up, _ = self.gate_up_proj(x)
         x = self.act_fn(gate_up)
-        x, _ = self.down_proj(x, can_fuse_mlp_allreduce=can_fuse_mlp_allreduce)
+        x, _ = self.down_proj(
+            x, skip_all_reduce=can_fuse_mlp_allreduce or use_reduce_scatter
```

Comment:
can we use `self.reduce_results` to control whether we want to skip the all reduce instead of introducing a new `skip_all_reduce`?

Reply:
No because we can only use reduce scatter if `forward_batch.dp_padding_mode.is_max_len()` is true which can change from batch to batch.

Reply:
I see. so, it seems reduce_results is a static attr of the layer but skip_all_reduce is more like a runtime param.

Reply:
Yes exactly